src/schemaCompare/schemaCompareWebViewController.ts

```diff
@@ -0,0 +1,323 @@
+} from "./schemaCompareUtils";
+import VscodeWrapper from "../controllers/vscodeWrapper";
+
+export class SchemaCompareWebViewController extends ReactWebviewPanelController<
+    SchemaCompareWebViewState,
+    SchemaCompareReducers
+> {
+    private operationId: string;
+
+    constructor(
+        context: vscode.ExtensionContext,
+        vscodeWrapper: VscodeWrapper,
+        node: any,
+        private readonly schemaCompareService: mssql.ISchemaCompareService,
+        private readonly connectionMgr: ConnectionManager,
+        schemaCompareOptionsResult: mssql.SchemaCompareOptionsResult,
+        title: string,
+    ) {
+        super(
+            context,
+            vscodeWrapper,
+            "schemaCompare",
+            "schemaCompare",
+            {
+                defaultDeploymentOptionsResult: schemaCompareOptionsResult,
+                sourceEndpointInfo: undefined,
+                targetEndpointInfo: undefined,
+                schemaCompareResult: undefined,
+                generateScriptResultStatus: undefined,
+                publishDatabaseChangesResultStatus: undefined,
+                schemaComparePublishProjectResult: undefined,
+                schemaCompareIncludeExcludeResult: undefined,
+                schemaCompareOpenScmpResult: undefined,
+                saveScmpResultStatus: undefined,
+                cancelResultStatus: undefined,
+            },
+            {
+                title: title,
+                viewColumn: vscode.ViewColumn.Active,
+                iconPath: {
+                    dark: vscode.Uri.joinPath(
+                        context.extensionUri,
+                        "media",
+                        "schemaCompare_dark.svg",
+                    ),
+                    light: vscode.Uri.joinPath(
+                        context.extensionUri,
+                        "media",
+                        "schemaCompare_light.svg",
+                    ),
+                },
+            },
+        );
+
+        this.operationId = generateOperationId();
+
+        if (!this.isTreeNodeInfoType(node)) {
+            node = this.getFullSqlProjectsPathFromNode(node);
+        }
+
+        void this.start(node);
+        this.registerRpcHandlers();
+    }
+
+    /**
+     * Starts the schema comparison process. Schema compare can get started with four contexts for the source:
+     * 1. undefined
+     * 2. Connection profile
+     * 3. Dacpac
+     * 4. Project
+     * @param sourceContext can be undefined, connection profile, dacpac, or project.
+     * @param comparisonResult Result of a previous comparison, if available.
+     */
+    public async start(
+        sourceContext: any,
+        comparisonResult: mssql.SchemaCompareResult = undefined,
+    ): Promise<void> {
+        let source: mssql.SchemaCompareEndpointInfo;
+
+        let connectionProfile: IConnectionProfile | undefined = sourceContext
+            ? <IConnectionProfile>sourceContext.ConnectionInfo
```

Comment:
We should use the new `myVar as IConnectionProfile` syntax over the old `<IConnectionProfile>myVar` to avoid conflicts with JSX/TSX syntax in case the code moves around.